### DISCUSSION TITLE
why cant we work with a single oracle??/

### DISCUSSION THREAD

**Discussion by egocentricfolklore16** [Cyfrin/foundry-full-course-cu#2273] [General]
why cant we work with single oracle

**Comment by EngrPips** [comment]
Because we have put all of our efforts, time, and resources into building a decentralized smart contract so, after all of those efforts, if we choose to work with a centralized oracle, then we have drive all our systems back to centralization as there is now a single point of failure for our whole system. Does this make sense to you?
